app/views/orders/_billing_form.html.erb

```diff
@@ -0,0 +1,7 @@
+<%= f.fields_for(:credit_card) do |ff| %>
+    <%= ff.input :billing_address1 %>
+    <%= ff.input :billing_address2 %>
+    <%= ff.input :billing_city %>
+    <%= ff.input :billing_state %>
+    <%= ff.input :billing_zip %>
```

Comment:
Don't take this as gospel but for the most part, you should avoid just naming variables as a single/double letter. It's understandable for sure but readability is more important than brevity. `form.field_for...` would always be more readable. But I don't know if this is something that's a big deal. I'll get back to this PR later today but I got meetings for awhile. Good so far! Just stylistic advice.